dataplane/saiserver/hostif_test.go

```diff
@@ -209,6 +218,112 @@ func TestRemoveHostif(t *testing.T) {
 	}
 }
 
+func TestCPUPacketStream(t *testing.T) {
+	dplane := &fakeSwitchDataplane{
+		ctx: fwdcontext.New("test", "test"),
+	}
+	p, err := fwdport.New(&fwdpb.PortDesc{
+		PortType: fwdpb.PortType_PORT_TYPE_CPU_PORT,
+		PortId:   &fwdpb.PortId{ObjectId: &fwdpb.ObjectId{Id: "2"}},
+		Port:     &fwdpb.PortDesc_Cpu{},
+	}, dplane.ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	c, mgr, stopFn := newTestHostif(t, dplane)
+	mgr.StoreAttributes(1, &saipb.SwitchAttribute{
+		CpuPort: proto.Uint64(2),
+	})
+
+	defer stopFn()
+
+	s, err := c.PacketIOClient.CPUPacketStream(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := s.Send(&pktiopb.PacketIn{}); err != nil {
+		t.Fatal(err)
+	}
+
+	t.Run("send", func(t *testing.T) {
+		if err := s.Send(&pktiopb.PacketIn{
+			Msg: &pktiopb.PacketIn_Packet{
+				Packet: &pktiopb.Packet{
+					Frame: []byte("hello"),
+				},
+			},
+		}); err != nil {
+			t.Fatal(err)
+		}
+		time.Sleep(10 * time.Millisecond)
+		want := [][]byte{[]byte("hello")}
+		if d := cmp.Diff(dplane.gotPackets, want); d != "" {
+			t.Errorf("PacketStream() failed: diff(-got,+want)\n:%s", d)
+		}
+	})
+	t.Run("recv", func(t *testing.T) {
+		if _, err := p.Write(testPacket(t, uint64(p.NID()))); err != nil {
+			t.Fatal(err)
+		}
+		got, err := s.Recv()
+		if err != nil {
+			t.Fatal(err)
+		}
+		want := &pktiopb.PacketOut{
+			Packet: &pktiopb.Packet{
+				HostPort:  1,
+				InputPort: 2,
+			},
+		}
+		if d := cmp.Diff(got, want, protocmp.Transform(), protocmp.IgnoreFields(&pktiopb.Packet{}, protoreflect.Name("frame"))); d != "" {
+			t.Errorf("PacketStream() failed: diff(-got,+want)\n:%s", d)
+		}
+	})
+}
+
+func testPacket(t testing.TB, nid uint64) fwdpacket.Packet {
```

Comment:
What's the purpose of this test? Look to me it only tests whether packet field update is successful?

Reply:
this is just a helper to create a packet, renamed func